src/HatsSignerGateBase.sol

```diff
@@ -527,10 +500,21 @@ abstract contract HatsSignerGateBase is BaseGuard, SignatureDecoder, HatsOwnedIn
         if (keccak256(abi.encode(owners)) != _existingOwnersHash) {
             revert SignersCannotChangeOwners();
         }
-        // prevent signers from changing the modules
+        // prevent signers from removing this module or adding any other modules
         /// @dev SENTINEL_OWNERS and SENTINEL_MODULES are both address(0x1)
-        (address[] memory modules,) = safe.getModulesPaginated(SENTINEL_OWNERS, enabledModuleCount + 1);
-        if (keccak256(abi.encode(modules)) != _existingModulesHash) {
+        (address[] memory modulesWith1, address next) = safe.getModulesPaginated(SENTINEL_OWNERS, 1);
+        // ensure that there is only one module...
+        if (
+            // if the length is 0, we know this module has been removed
+            // forgefmt: disable-next-line
+            modulesWith1.length == 0
+            /* per Safe ModuleManager.sol#137, "If all entries fit into a single page, the next pointer will be 0x1", ie SENTINEL_OWNERS.
+            Therefore, if `next` is not SENTINEL_OWNERS, we know another module has been added. */
+            || next != SENTINEL_OWNERS
+        ) {
+            revert SignersCannotChangeModules();
+        } // ...and that the only module is this contract
+        else if (modulesWith1[0] != address(this)) {
```

Comment:
This looks fine. You are checking three conditions with the same error message (and we exit the if early if modulesWith1.length == 0, so the indexing will work), so probably easiest to just stack them into one statement like:
```
if(
    modulesWith1.length == 0
    || next != SENTINEL_OWNERS
    || modulesWith1[0] != address(this)
) ...
```
Honestly, I would probably just get a page of 2 modules and simply check:
```solidity
if(modulesWith1.length != 1 || modulesWIth1[0] != address(this)) ...
```
But either way works. This solution will accomplish the goal, so marking it as approved.